fastapi_admin/routes/resources.py

```diff
@@ -86,28 +87,33 @@ async def update(
 ):
     form = await request.form()
     data, m2m_data = await model_resource.resolve_data(request, form)
-    async with in_transaction() as conn:
-        obj = (
-            await model.filter(pk=pk)
-            .using_db(conn)
-            .select_for_update()
-            .get()
-            .prefetch_related(*model_resource.get_m2m_field())
-        )
-        await obj.update_from_dict(data).save(using_db=conn)
-        for k, items in m2m_data.items():
-            m2m_obj = getattr(obj, k)
-            await m2m_obj.clear()
-            if items:
-                await m2m_obj.add(*items)
-        obj = (
-            await model.filter(pk=pk)
-            .using_db(conn)
-            .get()
-            .prefetch_related(*model_resource.get_m2m_field())
-        )
+    error = await model_resource.validate_data(request, data, m2m_data, pk=pk)
+    obj = None
+
+    if not error:
+        async with in_transaction() as conn:
+            obj = (
+                await model.filter(pk=pk)
+                .using_db(conn)
+                .select_for_update()
+                .get()
+                .prefetch_related(*model_resource.get_m2m_field())
+            )
+            await obj.update_from_dict(data).save(using_db=conn)
+            for k, items in m2m_data.items():
+                m2m_obj = getattr(obj, k)
+                await m2m_obj.clear()
+                if items:
+                    await m2m_obj.add(*items)
+            obj = (
+                await model.filter(pk=pk)
+                .using_db(conn)
+                .get()
+                .prefetch_related(*model_resource.get_m2m_field())
+            )
```

Comment:
I added this on my end to also pass Tortoise errors on edition and not just on creation
```suggestion
    if not error:
        try:
            async with in_transaction() as conn:
                obj = (
                    await model.filter(pk=pk)
                    .using_db(conn)
                    .select_for_update()
                    .get()
                    .prefetch_related(*model_resource.get_m2m_field())
                )
                await obj.update_from_dict(data).save(using_db=conn)
                for k, items in m2m_data.items():
                    m2m_obj = getattr(obj, k)
                    await m2m_obj.clear()
                    if items:
                        await m2m_obj.add(*items)
                obj = (
                    await model.filter(pk=pk)
                    .using_db(conn)
                    .get()
                    .prefetch_related(*model_resource.get_m2m_field())
                )
        except BaseORMException as exc:
            error = str(exc)
```